TideApp/TideApp/Domain/Network/NetworkManager.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  NetworkManager.swift
+//  TideApp
+//
+//  Created by John Sanderson on 19/05/2021.
+//
+
+import Foundation
+import Combine
+
+protocol NetworkManagerType {
+
+  func fetch<R: ResourceType>(_ resource: R) -> AnyPublisher<R.Output, Error>
+}
+
+final class NetworkManager: NetworkManagerType {
```

Comment:
This is just a slightly more generic network layer that could easily be used to make other requests

Reply:
Yeah thats was needed indeed! I would also think at some point to move the general network generic layer inside a module